# origin: Debian, #244585
# cset: n/a
# inclusion: not submitted
# description: Require firmware 2.70 on Alpha for DAC960
# revision date: 2004-09-04

diff -urN kernel-source-2.4.26-1/drivers/block/DAC960.c kernel-source-2.4.26-2/drivers/block/DAC960.c
--- kernel-source-2.4.26-1/drivers/block/DAC960.c	2002-11-30 15:49:38.000000000 +1100
+++ kernel-source-2.4.26-2/drivers/block/DAC960.c	2004-04-20 09:14:09.000000000 +1000
@@ -1133,6 +1133,26 @@
     DAC960PU/PD/PL	    3.51 and above
     DAC960PU/PD/PL/P	    2.73 and above
   */
+#if defined(CONFIG_ALPHA)
+  /*
+    DEC Alpha machines were often equipped with DAC960 cards that were
+    OEMed from Mylex, and had their own custom firmware. Version 2.70,
+    the last custom FW revision to be released by DEC for these older
+    controllers, appears to work quite well with this driver.
+
+    Cards tested successfully were several versions each of the PD and
+    PU, called by DEC the KZPSC and KZPAC, respectively, and having
+    the Manufacturer Numbers (from Mylex), usually on a sticker on the
+    back of the board, of:
+
+    KZPSC:  D040347 (1-channel) or D040348 (2-channel) or D040349 (3-channel)
+    KZPAC:  D040395 (1-channel) or D040396 (2-channel) or D040397 (3-channel)
+  */
+# define FIRMWARE_27X  "2.70"
+#else
+# define FIRMWARE_27X  "2.73"
+#endif
+
   if (Enquiry2.FirmwareID.MajorVersion == 0)
     {
       Enquiry2.FirmwareID.MajorVersion =
@@ -1152,7 +1172,7 @@
 	(Controller->FirmwareVersion[0] == '3' &&
 	 strcmp(Controller->FirmwareVersion, "3.51") >= 0) ||
 	(Controller->FirmwareVersion[0] == '2' &&
-	 strcmp(Controller->FirmwareVersion, "2.73") >= 0)))
+	 strcmp(Controller->FirmwareVersion, FIRMWARE_27X) >= 0)))
     {
       DAC960_Failure(Controller, "FIRMWARE VERSION VERIFICATION");
       DAC960_Error("Firmware Version = '%s'\n", Controller,

